conjure-go-client/httpclient/client.go

```diff
@@ -97,14 +97,16 @@ func (c *clientImpl) Do(ctx context.Context, params ...RequestParam) (*http.Resp
 	failedURIs := map[string]struct{}{}
 	for i := 0; i < c.maxRetries; i++ {
 		resp, err = c.doOnce(ctx, nextURI, params...)
-		if resp == nil {
-			// If we get a nil response, we can assume there is a problem with host and can move on to the next.
-			nextURI, offset = nextURIOrBackoff(nextURI, uris, offset, failedURIs, retrier)
-		} else if shouldThrottle, _ := internal.IsThrottleResponse(resp); shouldThrottle {
```

Comment:
IsThrottle response (and some adjacent functions) are no longer used now

Should we instead make `IsThrottleResponse` more "robust"
One concern I have is that there are consumers of this repo that are *not* using the rest error decoding middleware and that these will now be broken. If you can write `IsThrottleResponse`, and the others, to handle both cases I think we'll be better off.

Reply:
yep agreed, updating